etcdserver/apply.go

```diff
@@ -299,6 +310,9 @@ func (a *applierV3backend) Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error) {
 	if err := a.checkRequestRange(reqs); err != nil {
 		return nil, err
 	}
+	if err := a.checkRequestPermission(header, reqs); err != nil {
```

Comment:
also probably we should check the permission of compare/success/failure all together in one func if possible.